pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -602,10 +648,26 @@ public boolean commitSegmentMetadata(String rawTableName, final String committin
     return true;
   }
 
-  // TODO: Would be good to cache table config fields which are repeatedly checked
+  /**
+   * Helper function to return cached table config. It updates if the newer version of table config is observed.
+   *
+   * @param tableName name of the table
+   * @return table configuration that reflects the most recent version
+   */
+  private TableConfig getTableConfig(String tableName) {
```

Comment:
please write a more generic cache class like we discussed, and let us have only this class use the cache for now. We can expand that to be a static cache class later.